Snowflake.Data.Tests/IntegrationTests/SFBindTestIT.cs

```diff
@@ -769,5 +774,283 @@ public void testPutArrayBind1()
+        [TestCase(ResultFormat.ARROW, SFTableType.Iceberg, SFDataType.TIME, 6, DbType.Time, FormatHmsf, null)]
+        [TestCase(ResultFormat.ARROW, SFTableType.Iceberg, SFDataType.TIMESTAMP_NTZ, 6, DbType.DateTime, FormatYmdHms, null)]
+        // [TestCase(ResultFormat.ARROW, SFTableType.Iceberg, SFDataType.TIMESTAMP_TZ, 6, DbType.DateTime, FormatYmdHmsZ, null)] // Unsupported data type 'TIMESTAMP_TZ(6)' for iceberg tables
+        [TestCase(ResultFormat.ARROW, SFTableType.Iceberg, SFDataType.TIMESTAMP_LTZ, 6, DbType.DateTimeOffset, FormatYmdHmsZ, null)]
+        */
+        // Session TimeZone cases
+        [TestCase(ResultFormat.ARROW, SFTableType.Standard, SFDataType.TIMESTAMP_LTZ, 6, DbType.DateTimeOffset, FormatYmdHmsZ, "Europe/Warsaw")]
+        [TestCase(ResultFormat.JSON, SFTableType.Hybrid, SFDataType.TIMESTAMP_LTZ, 6, DbType.DateTimeOffset, FormatYmdHmsZ, "Asia/Tokyo")]
+        public void TestDateTimeBindingWithValue(ResultFormat resultFormat, SFTableType tableType, SFDataType columnType, Int32? columnPrecision, DbType bindingType, string comparisonFormat, string timeZone)
+        {
+            // Arrange
+            var timestamp = "2023/03/15 13:17:29.207 +05:00"; // 08:17:29.207 UTC
+            var expected = ExpectedTimestamp.From(timestamp, columnType);
+            var columnWithPrecision = ColumnTypeWithPrecision(columnType, columnPrecision);
+            var testInfo = $"TableType={tableType}, ColumnType={columnWithPrecision}, BindingType={bindingType}, ComparisonFormat={comparisonFormat}";
+            var bindingThreshold = 65280; // parameters limit for binding without bindings file on stage when exceeded enforces bindings file to stage
+            var batchSize = bindingThreshold/2; 
+            s_logger.Info(testInfo);
+            
+            using (IDbConnection conn = new SnowflakeDbConnection(ConnectionString))
+            {
+                conn.Open();
+
+                conn.ExecuteNonQuery($"alter session set DOTNET_QUERY_RESULT_FORMAT = {resultFormat}");
+                // conn.ExecuteNonQuery($"alter session set CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = {bindingThreshold}"); 
+                if (!timeZone.IsNullOrEmpty()) // Driver ignores this setting and relies on local environment timezone
+                    conn.ExecuteNonQuery($"alter session set TIMEZONE = '{timeZone}'");
+
+                CreateOrReplaceTable(conn, 
+                    TableName, 
+                    tableType.TableCreationPrefix(), 
+                    new[] { 
+                        "id number(10,0) not null primary key", // necessary only for HYBRID tables
+                        $"ts {columnWithPrecision}" 
+                    }, 
+                    tableType.TableCreationFlags());
+
+                // Act
+                var sqlInsert = $"insert into {TableName} (id, ts) values (?, ?)";
+                
+                // insert 1 row with value binding
```

Comment:
maybe lets try to extract each of the blocks into a method. If it turned out very complicated it may stay as is. But you could try.

Reply:
Done.